src/App.js

```diff
@@ -1,26 +1,86 @@
 import PubSub from 'pubsub-js';
-import { createBrowserHistory } from 'history';
 import memoize from 'lodash.memoize';
 import pick from 'lodash.pick';
 
 import getQueryParameters from './getQueryParameters';
-import { defaultAuthentication } from './frontendAuthWrapper';
-import validateConfig from './validateConfig';
+import { defaultAuthenticatedUser } from './frontendAuthWrapper';
+import * as handlers from './handlers';
 
 export const APP_TOPIC = 'APP';
+export const APP_BEFORE_INIT = `${APP_TOPIC}.BEFORE_INIT`;
+export const APP_CONFIGURED = `${APP_TOPIC}.CONFIGURED`;
+export const APP_VALIDATED = `${APP_TOPIC}.VALIDATED`;
+export const APP_AUTHENTICATED = `${APP_TOPIC}.AUTHENTICATED`;
+export const APP_I18N_CONFIGURED = `${APP_TOPIC}.I18N_CONFIGURED`;
+export const APP_LOGGING_CONFIGURED = `${APP_TOPIC}.LOGGING_CONFIGURED`;
+export const APP_ANALYTICS_CONFIGURED = `${APP_TOPIC}.ANALYTICS_CONFIGURED`;
+export const APP_BEFORE_READY = `${APP_TOPIC}.BEFORE_READY`;
```

Comment:
Is the APP_ prefix needed?

Reply:
Eh, just figured it'd be good to keep them nice and disambiguated from anything else.  `READY` and `ERROR` would be pretty vague constants.